packages/flutter/lib/src/services/sensitive_content.dart

```diff
@@ -0,0 +1,139 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// @docImport 'package:flutter/widgets.dart';
+library;
+
+import 'package:flutter/foundation.dart';
+
+import 'system_channels.dart';
+
+/// The level of sensitivity that content in the widget tree has.
+///
+/// {@template flutter.services.ContentSensitivity}
+/// There are only three [ContentSensitivity] levels, and these can be set via a
+/// [SensitiveContent] widget.
+///
+/// [ContentSensitivity.sensitive] is the most severe setting, and if it is set,
+/// it will cause the tree to remain marked sensitive even if there are other
+/// `SensitiveContent` widgets in the tree.
+///
+/// [ContentSensitivity.autoSensitive] is the second most severe setting, and it will
+/// cause the tree to remain marked auto-sensitive if there are either (1) no other
+/// `SensitiveContent` widgets in the tree or (2) there are only other auto-sensitive
+/// or not sensitive `SensitiveContent` widgets in the tree.
+///
+/// [ContentSensitivity.notSensitive] is the least severe setting, and it will cause the
+/// tree to remain marked not sensitive as long as there are (1) no other `SensitiveContent`
+/// widgets in the tree or (2) there  are only other not sensitive `SensitiveContent` widgets
+/// in the tree. If there are no `SensitiveContent` widgets in the tree, the default setting as
```

Comment:
I think you documentation is probably correct to the code but the code needs to change. The most likely reason someone sets "not sensitive" is because they want to explicitly force the streaming of pixels to work. 

We also want to default to safety so I understand why you authored it this way. 

My gut is that in dart we should say something like "Setting auto likely does nothing." And say that not overrides auto and that sensitive overrides not. Which still leaves us without the ideal way to override.

Will think more on this.

Reply:
I don't agree with making `notSensitive` a setting that can override others. Because we are implementing it as a widget and there can be multiple in the tree, when one is sensitive, I do think we must follow that for security's sake. I think this is an unfortunate consequence of the fact that we can not obscure widget by widget versus the entire screen.

I will work on rewording the `autoSensitive` method because it technically overrides `notSensitive` but behaviorally, does nothing.

Reply:
OK then how does someone turn off sensitivity for a debug build or dogfood build for their entire app while leaving the support around for their users.

Reply:
They can create a global variable that all the `SensitiveContent` widgets set as the `sensitivityLvel` in debug mode:

```dart
static ContentSensitivity debugModeContentSensitivity = ContentSensitivity.notSensitive;

SensitiveContent(
  sensitivityLevel: kDebugMode  ? debugModeContentSensitivity : myOtherContentSensitivitySetting,
  child: MyChildWidget(),
);

```